src/domain/collections/index.js

```diff
@@ -18,9 +18,51 @@ import {
 } from "../../components/table"
 import Button from "../../components/button"
 import Spinner from "../../components/spinner"
-import NewCollection from "./new"
+import AddCollectionModal from "../../components/templates/add-collection-modal"
+import { useAdminCreateCollection } from "medusa-react"
+import useToaster from "../../hooks/use-toaster"
+import { getErrorMessage } from "../../utils/error-messages"
 
 const CollectionsIndex = () => {
+  const [showNew, setShowNew] = useState(false)
+  const createCollection = useAdminCreateCollection()
+  const toaster = useToaster()
+  const { collections, isLoading, refresh, total_count } = useMedusa(
+    "collections",
+    {
+      search: `?${qs.stringify(filtersOnLoad)}`,
+    }
+  )
+
+  const handleAddNew = (data) => {
+    const payload = {
+      title: data.title,
+      handle: data.handle,
+    }
+
+    if (data.metadata) {
+      const metadata = data.metadata.reduce((acc, next) => {
+        return {
+          ...acc,
+          [next.key]: next.value,
+        }
+      }, {})
+
+      payload.metadata = metadata
+    }
+
+    createCollection.mutate(payload, {
+      onSuccess: () => {
+        toaster("Collection created", "success")
+        refresh()
```

Comment:
Should not be necessary, if we use the hook

Reply:
Agree but the collection view is being moved to the product overview in another PR and this domain will be deleted so this is just temporary 😄